src/basic/efivars.c

```diff
@@ -96,6 +96,17 @@ int efi_get_variable(
                                 (void) usleep_safe(EFI_RETRY_DELAY);
                 }
 
+                /* According to comment of linux source code, efivarfs represents uncommitted
+                 * variables with zero-length files. Reading them should return EOF, even though
+                 * st.st_size > 0
```

Comment:
Please link to #34304 in the comment too.

Also, "should ... even through st.st_size > 0" is wrong. This is still something for kernel to fix. Better not to spread the confusion even further, i.e. reword to be "Unfortunately kernel reports EOF if there's an inconsistency between efivarfs var list and what's actually stored in firmware"

Reply:
Okay.